src/messaging.cc

```diff
@@ -167,6 +176,43 @@ bool MessageEncoder::encode_array(Array* object, int size) {
   return true;
 }
 
+bool MessageEncoder::encode_map(Instance* instance) {
+  write_uint8(TAG_MAP);
+  Object* bk = instance->at(Instance::MAP_BACKING_OFFSET);
+  if (is_smi(bk)) return false;
+  HeapObject* backing = HeapObject::cast(bk);
+  Object* sz = instance->at(Instance::MAP_SIZE_OFFSET);
+  if (!is_smi(sz)) return false;
+  word size = Smi::cast(sz)->value();
+  write_cardinal(size);
+  if (size == 0) return true;  // Do this before looking at the backing, which may be null.
+  Smi* class_id = backing->class_id();
+  if (class_id == _program->list_class_id()) {
+    bk = Instance::cast(backing)->at(Instance::LIST_ARRAY_OFFSET);
```

Comment:
Maybe you should just replace bk and sz with object and use that consistently as a pattern -- or maybe we should have a few helpers on Instance to do this?